### DISCUSSION TITLE
group.addoption vs parser.addoption

### DISCUSSION THREAD

**Discussion by harmin-parra** [pytest-dev/pytest#11324] [Q&A] [answered]
Hi

What is the difference between:

`parser.addoption("my-option", ........)
`

and

```
group = parser.getgroup("my-plugin")
group.addoption("my-option", ........)
```

Does group allow different plugins to define or use the same option name ?
Does it allow an option name "collision" between plugins ?

----

Why don't we have the same thing for INI options ?

we have

`parser.addini("my-option", ........)`

but we don't have

`group.addini("my-option", ........)`

**Comment by RonnyPfannschmidt** [comment, marked as answer]
Group's simply categorize cli options for --help
